### ISSUE TITLE
Self-signed certificates are no longer working

### ISSUE THREAD

**Issue by romeoneverdies** [mozilla-mobile/fenix#17607]
self signed certificates are no longer working. 

the first time you visit a url you can add the exception but every-time after will fail stating connection is not secured. 

(should it not be up to me to decide who i trust?)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-9486)

**Comment by flowercow** [comment]
After the selfsigned cert is accepted by the old version ff, next connection will be fine. But now,  every new session of ff connection failed.
There should be a method that i can accept the selfsigned cert forever.

**Comment by majurgens** [comment]
It should just remember certificates that you have already accepted like firefox 68 mobile did and like firefox desktop does

**Comment by digitalcircuit** [comment]
### Additional context on why this matters
This affects anyone trying to use HTTPS in a Trust On First Use way with e.g. OpenWRT, FreshTomato, or even several OEM router firmwares (Verizon FiOS), alongside other Internet of Things devices (e.g. my Brother and HP printers).

Not only is it more inconvenient to have to approve each time, it's remarkably less secure to accept whatever certificate is shown to you *every* time, instead of being able to accept it once, then get alerted when it changes, as Firefox Fennec used to do and Firefox desktop currently does.

**Comment by quaintdev** [comment]
Any update on this? I'm hosting a dozen of apps at my home. For these apps I'm using self signed certs, every time I try to open these apps I have to keep accepting 'risk'.

**Comment by dcousens** [comment]
Without supporting exceptions for self-signed certificates, Fenix is encouraging users to use HTTP even when a secure HTTPS connection could be available.

Using the example of internal network services, like a webpage hosted on a `.home.arpa` or `.local` domain, you cannot visit the page using HTTPS without accepting the "risk" of a temporary exception.  As explained by @digitalcircuit,  routinely re-accepting this risk effectively negates any MITM security that the HTTPS certificate _could_ provide if exceptions were provided.

But not only that.

Rather than accept the pain of routinely accepting the "risk", users undoubtedly will simply switch to HTTP for some of these services than put up with that.  This is not only worse for the transport security when using the `.home.arpa` service, but insecure **HTTP is actually incompatible with new features like `HTTPS only` mode**.

If you visit the `.home.arpa` service with HTTPS only mode enabled, you are forced to wait for the connection to timeout before it reverts to HTTP.  The end result is users not using HTTPS when an exception could suffice; and potentially not using HTTPS opportunistically in normal browsing. :scream: 

##### In short
- Users that have to bypass HTTPS routinely might revert to using HTTP.
- HTTP is actively painful with HTTPS only mode enabled.
- Users then disable HTTPS only mode.

None of this is necessary, if Fenix simply (re)added support for self-signed HTTPS exceptions.

**Comment by gabrielluong** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813868

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
